manager/controllers/package_detail.go

```diff
@@ -0,0 +1,118 @@
+package controllers
+
+import (
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"github.com/gin-gonic/gin"
+	"github.com/jinzhu/gorm"
+	"net/http"
+)
+
+func singlePackageLatestQuery(pkgName string) *gorm.DB {
+	return database.Db.Table("package pkg").Select("*").Joins("JOIN package_name pn ON pkg.name_id = pn.id").
+		Joins("JOIN advisory_metadata adv ON pkg.advisory_id = adv.id").
+		Where("pn.name = ?", pkgName).
+		Order("adv.public_date DESC").Limit(1)
+}
+
+func singlePackageEvraQuery(pkgName string, evra string) *gorm.DB {
+	return database.Db.Table("package").
+		Joins("JOIN package_name ON package.name_id = package_name.id").
+		Where("package_name.name = ?", pkgName).
+		Where("package.evra = ?", evra)
+}
+
+func getAdvisoryByID(advID int) (models.AdvisoryMetadata, error) {
+	var advisory models.AdvisoryMetadata
+	err := database.Db.Where("id = ?", advID).First(&advisory).Error
+	return advisory, err
+}
+
+type PackageDetailResponse struct {
+	Data SinglePackage `json:"data"`
+	Name string        `json:"name"`
+}
+
+type SinglePackage struct {
+	Package  models.Package          `json:"package"`
```

Comment:
Please define special structs for response, PackageDetailItem, PackageDetailAttributes. Everything should be analogical to advisory_detail.go, do it consistently.